pkg/controller/lambda/function/setup.go

```diff
@@ -0,0 +1,499 @@
+	// We can't currently properly implement a comparison
+	return true
+}
+
+func isUpToDateLayers(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// The generator is creating layers with type of
+	// 	Layers []*string `json:"layers,omitempty"`
+	// Instead of []*svcsdk.Layer
+	// We can't properly implement a comparison until that is fixed
+	return true
+}
+
+func isUpToDateTracingConfig(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
+	mode := svcsdk.TracingModePassThrough
+	if !reflect.ValueOf(cr.Spec.ForProvider.TracingConfig).IsZero() &&
+		!reflect.ValueOf(cr.Spec.ForProvider.TracingConfig.Mode).IsZero() {
+		mode = *cr.Spec.ForProvider.TracingConfig.Mode
+	}
+	if !reflect.DeepEqual(mode, *obj.Configuration.TracingConfig.Mode) {
+		return false
+	}
+	return true
+}
+
+func isUpToDateSecurityGroupIDs(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
+	var securityGroupIDs []*string
+	var awsSecurityGroupIDs []*string
+	if !reflect.ValueOf(cr.Spec.ForProvider.VPCConfig).IsZero() &&
+		!reflect.ValueOf(cr.Spec.ForProvider.VPCConfig.SecurityGroupIDs).IsZero() {
+		securityGroupIDs = cr.Spec.ForProvider.VPCConfig.SecurityGroupIDs
+	}
+	if !reflect.ValueOf(obj.Configuration.VpcConfig).IsZero() &&
+		!reflect.ValueOf(obj.Configuration.VpcConfig.SecurityGroupIds).IsZero() {
+		awsSecurityGroupIDs = obj.Configuration.VpcConfig.SecurityGroupIds
+	}
+	// Check if counts match which should handle nil/empty
+	if len(securityGroupIDs) != len(awsSecurityGroupIDs) {
+		return false
+	}
+	// Check if actual values match, and only DeepEqual if there are actual values
+	if len(securityGroupIDs) > 0 &&
+		!reflect.DeepEqual(securityGroupIDs, awsSecurityGroupIDs) {
+		return false
+	}
+	return true
+}
+
+// returns which AWS Tags exist in the resource tags and which are outdated and should be removed
+func diffTags(spec map[string]*string, current map[string]*string) (map[string]*string, []*string) {
+	addMap := make(map[string]*string, len(spec))
+	removeTags := make([]*string, 0)
+	for k, v := range current {
+		if reflect.DeepEqual(spec[k], v) {
+			continue
+		}
+		removeTags = append(removeTags, aws.String(k))
+	}
+	for k, v := range spec {
+		if reflect.DeepEqual(current[k], v) {
+			continue
+		}
+		addMap[k] = v
+	}
+	return addMap, removeTags
+}
+
+type updater struct {
+	client svcsdkapi.LambdaAPI
+}
+
+func (u *updater) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*svcapitypes.Function)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#Lambda.UpdateFunctionCode
+	updateFunctionCodeInput := GenerateUpdateFunctionCodeInput(cr)
+	if _, err := u.client.UpdateFunctionCodeWithContext(ctx, updateFunctionCodeInput); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
```

Comment:
```suggestion
		return managed.ExternalUpdate{}, aws.Wrap(err, errUpdate)
```

All errors coming from AWS should be wrapped by our custom wrapper.